spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+RSpec.describe 'date selection', type: :feature do
+  include GdsApi::TestHelpers::ContentDataApi
+
+  before do
+    initial_page_stub
+  end
+
+  it 'renders data for the last 30 days if no date period is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 30.days
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+
+    expect_default_date_period_metrics_to_be_displayed(from, to)
+  end
+
+  it 'renders data for the last 30 days when `last 30 days` is selected' do
+    to = Time.zone.today
```

Comment:
Can you use a DateRange factory, using FactoryBot? You could use traits for it. 
This will dry-up the tests and will set a foundation for other tests to reuse it.